pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/IndexSegment.java

```diff
@@ -86,6 +87,15 @@ public interface IndexSegment {
    */
   GenericRow getRecord(int docId, GenericRow reuse);
 
+  /**
+   * Returns the primaryKey for a given docId
+   *
+   * @param docId Document Id
+   * @param reuse Reusable buffer for the primary key
+   * @return Primary key for the given document Id
+   */
+  void getPrimaryKey(int docId, PrimaryKey reuse);
```

Comment:
(minor) Consider changing `reuse` to `buffer` since the method has no return value

Reply:
I think we should keep this as reuse only since buffer has slightly different connotations.